src/devices/Tlc59711/Tlc59711.cs

```diff
@@ -0,0 +1,157 @@
+        /// <param name="channel">Channel, 0-based</param>
+        /// <param name="pwm">PWM level (0 = minimum, 65535 = maximum)</param>
+        public void SetPWM(int channel, ushort pwm)
+        {
+            if (channel > ChannelsPerDriver * _numberOfDrivers)
+            {
+                throw new ArgumentOutOfRangeException("channel");
+            }
+
+            _pwmBuffer[channel] = pwm;
+        }
+
+        /// <summary>
+        /// Call this to set the RGB value for a group of 3 channels
+        /// </summary>
+        /// <param name="ledNumber">LED number (channel number of the "red" pin divided by 3)</param>
+        /// <param name="red">Red level (0 = minimum, 65535 = maximum)</param>
+        /// <param name="green">Green level (0 = minimum, 65535 = maximum)</param>
+        /// <param name="blue">Blue level (0 = minimum, 65535 = maximum)</param>
+        public void SetLED(int ledNumber, ushort red, ushort green, ushort blue)
+        {
+            SetPWM((byte)(ledNumber * 3), red);
+            SetPWM((byte)(ledNumber * 3 + 1), green);
+            SetPWM((byte)(ledNumber * 3 + 2), blue);
+        }
+
+        /// <summary>
+        /// Call this after every change to write the new PWM levels to the device.
+        /// </summary>
+        public void Write()
+        {
+            int pos = 0;
+
+            uint command;
+
+            // Magic word for write
+            command = 0x25;
+
+            command <<= 5;
+            // OUTTMG = 1, EXTGCK = 0, TMGRST = 1, DSPRPT = 1, BLANK = 0 -> 0x16
+            command |= 0x16;
+
+            command <<= 7;
+            command |= _brightnessRed;
+
+            command <<= 7;
+            command |= _brightnessGreen;
+
+            command <<= 7;
+            command |= _brightnessBlue;
+
+            for (byte n = 0; n < _numberOfDrivers; n++)
+            {
+                _dataToWrite[pos++] = (byte)(command >> 24);
+                _dataToWrite[pos++] = (byte)(command >> 16);
+                _dataToWrite[pos++] = (byte)(command >> 8);
+                _dataToWrite[pos++] = (byte)(command);
+
+                // 12 channels per TLC59711
+                for (int c = ChannelsPerDriver - 1; c >= 0; c--)
+                {
+                    // 16 bits per channel, send MSB first
+                    _dataToWrite[pos++] = (byte)(_pwmBuffer[n * ChannelsPerDriver + c] >> 8);
+                    _dataToWrite[pos++] = (byte)(_pwmBuffer[n * ChannelsPerDriver + c]);
+                }
+            }
+
+            _spiDevice.Write(_dataToWrite);
+        }
+
+        /// <summary>
+        /// Set the brightness of all LED channels to same value
+        /// </summary>
+        /// <param name="brightness">Brightness between 0 and 127</param>
+        public void SetBrightness(byte brightness)
+        {
+            brightness = Math.Min(brightness, (byte)127);
+
+            _brightnessRed = _brightnessGreen = _brightnessBlue = brightness;
```

Comment:
how is this working? This won't change anything on the device, you're just updating fields here...